From 5c5905faf41bd0451ea54811fa6d94910dbc9128 Mon Sep 17 00:00:00 2001
From: Geetha sowjanya <gakula@marvell.com>
Date: Fri, 12 Apr 2019 16:57:30 +0530
Subject: [PATCH 0157/1921] octeontx2-vf: Configure backpressure level for
 packet reception

Enables and sets receive side backpressure levels for NIX CQ and
NPA Aura so that upon exhaustion of resources, CGX is backpressured
and sends out pause/PFC frames to transmitter.

Change-Id: I452a543b492147e19bbebbfa9fa8fc8fe0324577
Signed-off-by: Geetha sowjanya <gakula@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/nic/otx2_common.c | 1 +
 drivers/net/ethernet/marvell/octeontx2/nic/otx2_vf.c     | 3 +++
 2 files changed, 4 insertions(+)

diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_common.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_common.c
index f1e35c2efe4b..b8b94d5600a6 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_common.c
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_common.c
@@ -1312,6 +1312,7 @@ void mbox_handler_nix_bp_enable(struct otx2_nic *pfvf,
 		pfvf->bpid[chan_id] = rsp->chan_bpid[chan] & 0x3FF;
 	}
 }
+EXPORT_SYMBOL(mbox_handler_nix_bp_enable);
 
 void otx2_free_cints(struct otx2_nic *pfvf, int n)
 {
diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_vf.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_vf.c
index 872cc6749584..b366c9eb6cf8 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_vf.c
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_vf.c
@@ -78,6 +78,9 @@ static void otx2vf_process_vfaf_mbox_msg(struct otx2_nic *vf,
 		mbox_handler_nix_txsch_alloc(vf,
 					     (struct nix_txsch_alloc_rsp *)msg);
 		break;
+	case MBOX_MSG_NIX_BP_ENABLE:
+		mbox_handler_nix_bp_enable(vf, (struct nix_bp_cfg_rsp *)msg);
+		break;
 	default:
 		if (msg->rc)
 			dev_err(vf->dev,
-- 
2.31.1

